src/main/java/seedu/mark/logic/commands/TabCommand.java

```diff
@@ -0,0 +1,54 @@
+package seedu.mark.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.mark.commons.core.index.Index;
+import seedu.mark.logic.commands.commandresult.CommandResult;
+import seedu.mark.logic.commands.commandresult.TabCommandResult;
+import seedu.mark.model.Model;
+
+public class TabCommand extends Command {
+
+    public static final String COMMAND_WORD = "tab";
+
+    public static final String MESSAGE_SWITCH_ACKNOWLEDGEMENT = "Switching view as requested.";
+
+    public static final String MESSAGE_USAGE =
+            COMMAND_WORD + ": Switches view to the tab identified by the index"
+            + "given. Dashboard tab is 1, Online tab is 2, Offline tab is 3.\n"
+            + "Parameters: INDEX (must be either 1, 2 or 3)\n"
+            + "Example: " + COMMAND_WORD + " 1 ";
+
+    private static final int DASHBOARD = 1;
+    private static final int ONLINE = 2;
+    private static final int OFFLINE = 3;
```

Comment:
Suggestion: create a `Tab` enum? Then `TabCommandParser` can parse the index number to a Tab before calling the constructor `TabCommand(Tab)`

Reply:
I'm quite unsure whether `TabCommand` should handle the specifics or `TabCommandParser`...
As you mentioned earlier, taking `EditCommand` as reference, the check for whether, given a potentially valid index, that index is indeed valid for the particular command or not. If we do the same for `TabCommand`, then `TabCommandParser` should pass `TabCommand` the `index`.
But then again the it makes sense for the Parser to handle everything too...
What do you think?

Reply:
Hmm I see the trouble. How about creating a new method `parseTab(String)` in `ParserUtil` that creates a `Tab` from a given input?

Reply:
Oh sounds like a plan! Thanks I'll get it done